text/3371-cargo-target-directories.md

```diff
@@ -239,6 +254,12 @@ solve some of these or ensure forward compatibility as well as the solution prop
 
 On the other hand, `targo` is already here and working for at least one person, making it the most viable alternative for now.
 
+### Providing backlinks
+
+`targo` provides backlink (it links from it's own target directory to `<workspace>/target`) as a way for existing tools to continue working despite there being no `CARGO_TARGET_DIR` set for them to find the real target dire.
+
+`cargo` does not for `CARGO_TARGET_DIR` and will not do it either for `CARGO_TARGET_DIRECTORIES` : it provides `cargo metadata` that is the blessed way to obtain the actual target directory and when `CARGO_TARGET_DIRECTORIES` is used, it will set `CARGO_TARGET_DIR` on all invocation (if not already set) to make it easy to obtain the target directory for simple task (e.g. a test needing to launch another binary in the repo).
```

Comment:
Are backlinks for existing tools to work or to allow new tooling and users to understand what `<name>-<hash>` is meant to be serving (e.g. this would allow cargo-clean-sweep to clean up directories that are no longer referenced)

Reply:
I assume you're asking about `targo`'s backlink ? They're here to keep regular `cargo` command working since there is `CARGO_TARGET_DIR` set in such a case so `cargo` doesn't know where to find the true `target` dir and will just create a new one locally if it's not there

Reply:
I just confirmed with @sunshowers that backlinks (and "last updated" time stamps) exist to allow garbage collection of the base directory since people will create and delete projects without running `cargo clean`, leaking target directories.

Reply:
The discussion on backlinks within the RFC is still focused on compatibility (which I don't think is even a thing) and does not cover GC

Reply:
btw we are considering a more general GC scheme within cargo: https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/last-use.20tracking

Reply:
FYI this still needs to be addressed

Reply:
> does not cover GC

This is what you want addressed right ?

I don't think there is anything to address. `cargo` has zero garbage collection currently (of old artifacts from previous rust versions) for example

I have ideas about using the unspecified naming scheme to have something like `<hash>/rustc-1.X.Y/....`, which would allow for a command like `cargo clean --old` which would remove artifacts from older versions in `CARGO_TARGET_BASE_DIR` but I don't think this is necessary for this RFC

Reply:
The motivation listed for backlinks is incorrect.  It is not for any "compatibility" but so that GC can be added later.

I think its also worth mentioning GC and I've called that out separately to be added as a future possibility

Reply:
I went ahead and fixed the entry on backlinks in 0e3e6d1b